services/FileService.java

```diff
@@ -278,10 +278,9 @@ private static String getDateTimeUtc() {
     private static void addFormField(DataOutputStream out, String name, String value) throws IOException {
         out.writeBytes(TWO_HYPHENS + BOUNDARY + CRLF);
         out.writeBytes("Content-Disposition: form-data; name=\"" + name + "\"" + CRLF);
-        if (value != null) {
-            out.writeBytes(CRLF);
-            out.writeBytes(value + CRLF);
-        }
+        if (value == null)  value = "";
+        out.writeBytes(CRLF);
+        out.writeBytes(value + CRLF);
```

Comment:
I think the least invasive and most clean way is to put it like this: 

out.writeBytes(CRLF);
if (value != null) {
    out.writeBytes(value);
}
out.writeBytes(CRLF);

With the first write, you finish the headers. As seen from the code in the apache lib used by SC, it always tries to read 2 crlfs as a marker for the end of the headers. Then you write the 'value' if it is not null, and finally, you write the last CRLF which testing showed is actually necessary even when the 'value' is null. 

Remove the assignment of 'value' to an empty string ("") and also the passing of "" argument in the calling function.